src/libraries/System.Memory.Data/ref/System.Memory.Data.cs

```diff
@@ -10,24 +10,33 @@ namespace System
     public partial class BinaryData
     {
         public BinaryData(byte[] data) { }
+        public BinaryData(byte[] data, string? mediaType) { }
         [System.Diagnostics.CodeAnalysis.RequiresDynamicCodeAttribute("JSON serialization and deserialization might require types that cannot be statically analyzed and might need runtime code generation.")]
         [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("JSON serialization and deserialization might require types that cannot be statically analyzed.")]
         public BinaryData(object? jsonSerializable, System.Text.Json.JsonSerializerOptions? options = null, System.Type? type = null) { }
         public BinaryData(object? jsonSerializable, System.Text.Json.Serialization.JsonSerializerContext context, System.Type? type = null) { }
         public BinaryData(System.ReadOnlyMemory<byte> data) { }
+        public BinaryData(System.ReadOnlyMemory<byte> data, string? mediaType) { }
         public BinaryData(string data) { }
+        public BinaryData(string data, string? mediaType) { }
         public static System.BinaryData Empty { get { throw null; } }
+        public string? MediaType { get { throw null; } }
```

Comment:
I would like to propose that this be named ContentType rather than MediaType.
See https://github.com/dotnet/runtime/issues/87068#issuecomment-1599661542
/cc @KrzysztofCwalina

Reply:
We have a complex ContentType class that contains MediaType property inside https://learn.microsoft.com/dotnet/api/system.net.mime.contenttype

Reply:
Ok fair enough.